linux/device.go

```diff
@@ -495,6 +515,54 @@ func createLinuxDevice(volume *model.Volume) (dev *model.Device, err error) {
 			// Match SerialNumber
 			if d.SerialNumber == volume.SerialNumber {
 				log.Debugf("Found device with matching SerialNumber:%s map %s and slaves %+v", d.SerialNumber, d.AltFullPathName, d.Slaves)
+
+				if volume.EncryptionKey != ""{
+					originalDevPath := "/dev/" + d.Pathname
+
+					isLuksDev, err := isLuksDevice(originalDevPath)
+					if err != nil {
+						return nil, err
+					}
+
+					// LUKS format device if this is the first time it is being used
+					if !isLuksDev {
+						log.Infof("Device %s is a new device. LUKS formatting it...", originalDevPath)
+						formatCmd := exec.Command("cryptsetup", "luksFormat", "--batch-mode", originalDevPath)
+						log.Debugf("luksFormat command:%s", formatCmd)
+						formatCmd.Stdin = strings.NewReader(volume.EncryptionKey)
+						if err := formatCmd.Run(); err != nil {
```

Comment:
util.ExecCommandOutput() should have been used here but cryptsetup requires reading passphrase from stdin which at the moment is not supported by util.ExecCommandOutput(). This is a work in progress.

Reply:
is there a way to pass it as an argument ?

Reply:
The only way to pass it as an argument seems to be via a file containing the passphrase. However, it seems to be an overkill to create an temporary file on the fly and then to delete it once the job is done.